src/v/kafka/server/connection_context.cc

```diff
@@ -132,6 +132,10 @@ ss::future<> connection_context::process_one_request() {
           conn->addr,
           e.what());
         conn->shutdown_input();
+    } catch (const sasl_session_expired_exception& e) {
+        vlog(
+          klog.warn, "SASL session expired for {} - {}", conn->addr, e.what());
+        conn->shutdown_input();
```

Comment:
question: Should there be a new metric (just a count) of the number of _expired_ authn connections?

Reply:
Yeah probably. There's a [task](https://github.com/redpanda-data/core-internal/issues/802) for OIDC metrics, but this is a bit more general. Can add quickly.